cloud-sql-proxy/cloud-sql-proxy.sh

```diff
@@ -214,6 +226,50 @@ EOF
 
 }
 
+
+# Helper to run any command with default bdutil settings with Fibonacci
+# backoff. If all retries fail, returns last attempt's exit code.
+# Args: "$@" is the command to run.
+function run_with_retries() {
+  local retry_backoff=(1 1 2 3 5 8 13 21 34 55 89 144)
+  local -a cmd=("$@")
+  echo "About to run '${cmd[*]}' with retries..."
+
+  local update_succeeded=0
+  for ((i = 0; i < ${#retry_backoff[@]}; i++)); do
+    if "${cmd[@]}"; then
+      update_succeeded=1
+      break
+    else
+      local sleep_time=${retry_backoff[$i]}
+      echo "'${cmd[*]}' attempt $(( $i + 1 )) failed! Sleeping ${sleep_time}." >&2
+      sleep ${sleep_time}
+    fi
+  done
+
+  if ! (( ${update_succeeded} )); then
+    echo "Final attempt of '${cmd[*]}'..."
+    # Let any final error propagate all the way out to any error traps.
+    "${cmd[@]}"
+  fi
+}
+
+
+function configure_hive_warehouse_dir(){
+  run_with_retries nc -zv "${MASTER_HOSTNAMES[0]}" 3306
+
+  HIVE_WAREHOURSE_URI=$(beeline -u jdbc:hive2://localhost:10000 -e "describe database default;" 2>/dev/null \
+    | sed '4q;d' | cut -d "|" -f4 | tr -d '[:space:]')
```

Comment:
Wondering if there is a more elegant way to do this - describe database default gives you a table of info about the database - maybe you could use the table as input and do select location from (the table)

Reply:
Checked, it's only possible to query the table in metastore db, in hive server, the meta data is not stored in any table.